### DISCUSSION TITLE
Automate creation of repository and branches

### DISCUSSION THREAD

**Discussion by vamshidharkibm** [community/community#21775] [New to GitHub]
<p>I would like to know, is there any possibility of automated scripts, which will help us to setup the Repository and branches specified?<br>
For example,<br>
A script that can be executed by reading following information from a properties file and create a repository named “MyProject” and also create five branches named dev1, dev2, test, preprod, prod.</p>
<h2>git_repo_setup.properties:</h2>
<h2>repo=MyProject<br>
branches=dev1, dev2, test, preprod, prod</h2>
<p>Also, be able to push the base setup files into “master” branch. Including the support for Large File size<br>
Please check and let me know if this is possible.<br>
Thanks<br>
Vamshi</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/vamshidharkibm">@vamshidharkibm</a>,</p>
<p>Thank you for being here!  It is possible to automate the creation of repositories by using one of the two <a href="https://developer.github.com/enterprise/2.11/v3/repos/#create" rel="nofollow noopener"><code>create</code> API end points</a>.</p>
<p>Depending on your use case, you may also find <a href="https://developer.github.com/enterprise/webhooks/" rel="nofollow noopener">webhooks</a> useful. For instance, you could use a webhook to create a new repository whenever an issue is created requesting one.</p>

**Comment by vamshidharkibm** [comment]
<p>Thank you <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> </p>
<p>I’m new to REST APIs; Can anyone provide a way how can I use them?</p>
<p>Can I use them from a Java Program or does it work only shell script?</p>
<p>Is there any sample code snippet?</p>
<p>Below is what I tried from Terminal, but branch is not created</p>
<hr>
<p>curl -i -H “Authorization: token &lt;my_token&gt;” \<br>
-d ‘{ \<br>
“name”: “my_branch_name”, \<br>
“auto_init”: true, \<br>
“private”: true, \<br>
“gitignore_template”: “nanoc” \<br>
}’ \<br>
<a href="https://github.com/" rel="nofollow noopener">https://github.com/</a>&lt;mygit&gt;/&lt;myrepo&gt;git</p>
<hr>
<p>Thanks</p>
<p>Vamshi</p>

**Comment by vamshidharkibm** [comment]
<p>Hi,</p>
<p>I’m able to use it in the following way</p>
<p>curl -u $gitUserName <a href="https://api.github.com/user/repos" rel="nofollow noopener">https://api.github.com/user/repos</a> -d “{“name”:”$repoName"}"</p>
<p>Thanks</p>
<p>Vamshi</p>

**Comment by vamshidharkibm** [comment]
<p>Hi,</p>
<p>The below command from the shell script is working fine on my Red Hat Enterprise Linux. But the same does not work from the git bash env. on a Windows system.</p>
<p>The control hangs at this line from the shell script.</p>
<p>What could be the issue?</p>
<p>curl -u $gitUserName <a href="https://api.github.com/user/repos" rel="nofollow noopener">https://api.github.com/user/repos</a> -d “{“name”:”$repoName"}"</p>
<p>Thanks</p>
<p>Vamshi</p>

**Comment by vamshidharkibm** [comment]
<p>It works now with the following</p>
<p><strong>winpty curl</strong></p>
<p>Thanks</p>
<p>Vamshi</p>
